packages/vuetify/src/components/VInfiniteScroll/VInfiniteScroll.tsx

```diff
@@ -0,0 +1,257 @@
+// Styles
+import './VInfiniteScroll.sass'
+
+// Components
+import { VBtn } from '@/components/VBtn'
+import { VProgressCircular } from '@/components/VProgressCircular'
+
+// Composables
+import { makeDimensionProps, useDimension } from '@/composables/dimensions'
+import { useIntersectionObserver } from '@/composables/intersectionObserver'
+import { useLocale } from '@/composables/locale'
+
+// Utilities
+import { computed, nextTick, onMounted, ref, watch } from 'vue'
+import { convertToUnit, defineComponent, useRender } from '@/util'
+
+// Types
+import type { PropType } from 'vue'
+
+export type InfiniteScrollSide = 'start' | 'end' | 'both'
+export type InfiniteScrollStatus = 'ok' | 'empty' | 'loading' | 'error'
+
+const VInfiniteScrollIntersect = defineComponent({
+  name: 'VInfiniteScrollIntersect',
+
+  props: {
+    side: {
+      type: String as PropType<InfiniteScrollSide>,
+      required: true,
+    },
+    rootRef: null,
+    rootMargin: String,
+  },
+
+  emits: {
+    intersect: (side: InfiniteScrollSide) => true,
+  },
+
+  setup (props, { emit }) {
+    const { intersectionRef, isIntersecting } = useIntersectionObserver(entries => {
+    }, props.rootMargin ? {
+      root: props.rootRef,
+      rootMargin: props.rootMargin,
+    } : undefined)
+
+    watch(isIntersecting, async val => {
+      if (val) emit('intersect', props.side)
+    })
+
+    useRender(() => (
+      <div class="v-infinite-scroll-intersect" ref={ intersectionRef }>&nbsp;</div>
+    ))
+  },
+})
+
+export const VInfiniteScroll = defineComponent({
+  name: 'VInfiniteScroll',
+
+  props: {
+    color: String,
+    load: {
```

Comment:
What if we made this optional and instead provide the user with 2 more pieces of functionality?

1. emit the load event
2. add a loading prop

Reply:
I'm not very keen on having multiple ways of interacting with the component. But I'm not against finding an alternative to the current solution. However just having load event and loading prop would not be enough to replace what is there now.

Both sides of the infinite list have independent statuses, which include not only `loading` but `error` and `empty` as well.